pkg/registry/clusterrolebinding/policybased/storage.go

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
@erictune after more thinking, I think the inherent promise that role bindings don't apply at the cluster level gets us most of this behavior we want.

The mutability of roles, means we have to address @olegshaldybin's comments. And in this case, I'm having trouble figuring out how to centralize all of this logic since there's no good example of existing storage validation requiring API access.

Making these checks an admission controller means we have to deal with @liggitt's comments of unintentional escalation, but it also gets us the localhost without escalation checks we talked about earlier.

Wondering if there's some way we can deal with this after we merge the API group.